### ISSUE TITLE
Need support to define the new cloudfront function through the serverless yml

### ISSUE THREAD

**Issue by noobyblob** [serverless-nextjs/serverless-next.js#1748]
**Is your feature request related to a problem? Please describe.**
I have been using the new AWS cloudfront functions for some utilities and I am able to author the function's association to my aws distribution using the aws console. It works great except that when I do a follow up serverless deployment, all the function existing associations get removed.

**Describe the solution you'd like**
I would like a cleaner way to prevent overriding the cloudfront function associations or need an easy way to define those associations through yml.

**Describe alternatives you've considered**
I have been doing a manual override of function associations after every final deployment.

**Additional context**
More info on the new aws cloud front function feature can be found at https://aws.amazon.com/blogs/aws/introducing-cloudfront-functions-run-your-code-at-the-edge-with-low-latency-at-any-scale/

**Comment by dphang** [comment]
Yes I think we hadn't considered CloudFront functions yet so the component code might be doing something like overriding any function associations.

If you are comfortable you can try to take a look around here https://github.com/serverless-nextjs/serverless-next.js/blob/cf5bd5d023516f0269b35098d0642c5c7f546bae/packages/serverless-components/nextjs-component/src/component.ts#L909-L909 to figure out where it might be overriding, basically here is where we set the behaviors.

**Comment by chorkaichan** [comment]
I'm also using cloudfront functions, and facing this issue. I'll see if I can fix it.
